src/server/worker/api_server.go

```diff
@@ -409,7 +423,20 @@ func (a *APIServer) downloadGitData(pachClient *client.APIClient, dir string, in
 
 func (a *APIServer) downloadData(pachClient *client.APIClient, logger *taggedLogger, inputs []*Input, puller *filesync.Puller, parentTag *pfs.Tag, stats *pps.ProcessStats, statsTree hashtree.OpenHashTree, statsPath string) (_ string, retErr error) {
 	defer func(start time.Time) {
-		stats.DownloadTime = types.DurationProto(time.Since(start))
+		duration := time.Since(start)
+		stats.DownloadTime = types.DurationProto(duration)
+		if a.exportStats {
+			if hist, err := datumDownloadTime.GetMetricWithLabelValues(a.pipelineInfo.ID, a.jobID); err != nil {
+				logger.Logf("failed to get histogram w labels: pipeline (%v) job (%v) with error %v", a.pipelineInfo.ID, a.jobID, err)
+			} else {
+				hist.Observe(duration.Seconds())
+			}
+			if counter, err := datumDownloadSecondsCount.GetMetricWithLabelValues(a.pipelineInfo.ID, a.jobID); err != nil {
+				logger.Logf("failed to get counter w labels: pipeline (%v) job (%v) with error %v", a.pipelineInfo.ID, a.jobID, err)
+			} else {
+				counter.Add(duration.Seconds())
+			}
+		}
```

Comment:
How would you feel about factoring this into its own function? I feel like this defer block is getting big enough that it impacts readability of `downloadData`, would be nice if it just read `defer downloadMetrics(time.Now())` or smth like that.